clang/lib/CConv/RewriteUtils.cpp

```diff
@@ -1225,7 +1225,44 @@ class CastPlacementVisitor :
   }
 
   bool VisitBinAssign(BinaryOperator *O) {
-    // TODO: Aron try to add cast to RHS expression..if applicable.
+    // This is an assignment statement.
```

Comment:
This is a lot of code replacing something that was simple. Can you explain?

Reply:
I implemented the feature, which was not there before. In few cases, we may need to insert fancy cast, especially when we assign from a complex `&` expression i.e., `p = &foo->bar[d];` `clang` complains that we are trying to assign non-checked pointers to checked pointers. Similarly, for explicit casts i.e., `p = (int*)foo(a.b);` we should add a fancy cast to the RHS.

Reply:
But, I am wondering if this fix is in the wrong place. Why are we not inferring the correct type for `&foo->bar[d]` so that we don't have to insert a cast? Is the cast a band-aid over a different problem?

Reply:
Yes, it should be handled properly by clang.but clang throws an error. Lets raise an issue with mainline clang and see what they say

Reply:
OK ... But I'd still like to see a test case for this, so we can explore if there might be other solutions (or problems we should also fix).